### PR TITLE
Feat/update user info

### PR DESCRIPTION
**Pull request by Sgalvin36**
## Type of Change
- [x] feature ⛲
- [x] documentation update 📃
- [x] testing 🧪

## Description

Added the update user functionality to the user controller. Provided tests for that within the requests folder. Updated the readme to show the new route availability.

## Motivation and Context

[Issue 8](https://github.com/turingschool/tracker-crm/issues/8)

## Related Tickets

Resolves half the ticket. The back-end component of the ticket should be complete with this update.

## Added Test?
- [x] Yes 🫡
  - 
  - Request/Controller specs
- [ ] No 🙅
- [ ] All previous tests still pass 🥳

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by jimmacur** [comment]
Your additions to the documentation are very helpful and thorough. Update endpoint in controller is looking clean and uses helper method to match the style of the controller. Tests are thorough including happy path and sad path for non-unique email, bad password, and missing field. Looks ready to merge.

**Review by jimmacur — COMMENTED** [review]
Your additions to the documentation are very helpful and thorough. Update endpoint in controller is looking clean and uses helper method to match the style of the controller. Tests are thorough including happy path and sad path for non-unique email, bad password, and missing field. Looks ready to merge.

**Review by jimmacur — APPROVED** [review]
Your additions to the documentation are very helpful and thorough. Update endpoint in controller is looking clean and uses helper method to match the style of the controller. Tests are thorough including happy path and sad path for non-unique email, bad password, and missing field. Looks ready to merge.

File: app/controllers/api/v1/users_controller.rb [hunk 1]
```diff
@@ -18,6 +18,16 @@ def show
         render json: UserSerializer.new(User.find(params[:id]))
       end
 
+      def update
+        user = User.find(params[:id])
+        user.assign_attributes(user_params)
```
Comment (kylomite) [inline app/controllers/api/v1/users_controller.rb:23]: I want to learn more about where `assign_attributes` is coming from. It looks like its 
- updating `self`
- saving the change to self
Do I have that correct?

**Review by kylomite — APPROVED** [review]

**Pull request merged**